common/src/main/java/com/microsoft/identity/common/internal/broker/ipc/ContentProviderStrategy.java

```diff
@@ -0,0 +1,132 @@
+
+/**
+ * A strategy for communicating with the targeted broker via Content Provider.
+ */
+public class ContentProviderStrategy implements IIpcStrategy {
+
+    private static final String TAG = ContentProviderStrategy.class.getName();
+    private final Context mContext;
+
+    public ContentProviderStrategy(final Context context) {
+        mContext = context;
+    }
+
+    @Override
+    @Nullable public Bundle communicateToBroker(@NonNull BrokerOperationBundle brokerOperationBundle)
+            throws BaseException {
+
+        final String methodName = brokerOperationBundle.getOperation().name();
+
+        final Uri uri = getContentProviderURI(
+                brokerOperationBundle.getTargetBrokerAppPackageName(),
+                brokerOperationBundle.getContentProviderPath()
+        );
+        Logger.info(TAG + methodName, "Request to BrokerContentProvider for uri path " +
+                brokerOperationBundle.getContentProviderPath()
+        );
+
+        String marshalledRequestString = null;
+
+        final Bundle requestBundle = brokerOperationBundle.getBundle();
+        if (requestBundle != null) {
+            byte[] marshalledBytes = ParcelableUtil.marshall(requestBundle);
+            marshalledRequestString = Base64.encodeToString(marshalledBytes, 0);
+        }
+
+        final Cursor cursor = mContext.getContentResolver().query(
+                uri,
+                null,
+                marshalledRequestString,
+                null,
+                null
+        );
+
+        if (cursor != null) {
+            final Bundle resultBundle = cursor.getExtras();
+            cursor.close();
+            Logger.info(TAG + methodName, "Received successful result from Broker Content Provider.");
+            return resultBundle;
+        } else {
+            final String message = "Failed to get result from Broker Content Provider, cursor is null";
+            Logger.error(TAG + methodName, message, null);
+            throw new BrokerCommunicationException(message, null);
+        }
+    }
+
+    /**
+     * Returns a content provider URI for the given path.
+     */
+    private static Uri getContentProviderURI(@NonNull final String activeBrokerPackageName,
+                                             @NonNull final String path) {
+        final String authority = getContentProviderAuthority(activeBrokerPackageName);
+        return Uri.parse(CONTENT_SCHEME + authority + path);
+    }
+
+    /**
+     * Returns content provider authority.
+     */
+    private static String getContentProviderAuthority(@NonNull final String activeBrokerPackageName) {
+        return activeBrokerPackageName + "." + AUTHORITY;
+    }
+
+    /**
+     * Returns true of the target package name supports this content provider strategy.
+     */
+    public static boolean isBrokerContentProviderAvailable(@NonNull final Context applicationContext,
```

Comment:
It would be convenient to make these non-static, and just configure with a list of them.